dao/src/test/java/com/fitgoal/dao/impl/UserDaoImplTest.java

```diff
@@ -0,0 +1,120 @@
+package com.fitgoal.dao.impl;
+
+import com.fitgoal.dao.UserDao;
+import com.fitgoal.dao.domain.UserDto;
+import com.fitgoal.dao.util.MySqlTestHelper;
+import org.apache.ibatis.io.Resources;
+import org.apache.ibatis.session.SqlSessionFactory;
+import org.apache.ibatis.session.SqlSessionFactoryBuilder;
+import org.apache.ibatis.session.SqlSessionManager;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.testcontainers.containers.MySQLContainer;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.Properties;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class UserDaoImplTest {
+
+    @ClassRule
+    public static MySQLContainer mySQLContainer = (MySQLContainer) new MySQLContainer("mysql:latest")
+            .withDatabaseName("test_db")
+            .withInitScript("sqlScripts/create-db-schema.sql");
+
+    private static UserDao dao;
+    private static MySqlTestHelper testHelper;
+
+    @BeforeClass
+    public static void setUpSessionManager() {
+        SqlSessionManager sessionManager = SqlSessionManager.newInstance(getSqlSessionFactory());
+        dao = new UserDaoImpl(sessionManager);
+        testHelper = new MySqlTestHelper(mySQLContainer);
+    }
+
+    @After
+    public void deleteAllFromDBAfterEachTest(){
+        testHelper.deleteAll("users");
+    }
+
+    @Test
+    public void whenCreateUserAndGetByEmail_thenReturnUser() {
```

Comment:
the pattern for test naming : operation_whenXXX_assert/expect/return/throwXXXX.
In this case it would be findByEmail_whenUserExists_assertContent

Reply:
Done